### PR TITLE
Add a parameter TestPriority, to enable run test by priority

### PR DESCRIPTION
**Pull request by xian123**
1 Fix the bug of update the value of TestCategory,TestArea,TestNames and TestTag failed
2 Add a parameter TestPriority, when it is enabled, cases exact equal priorities will be run

### CODE REVIEW

**Comment by lizzha** [comment]
@xian123 please resolve the conflict

File: Libraries/XmlProcessing.psm1 [hunk 1]
```diff
@@ -30,10 +30,6 @@ Function Import-TestCases($WorkingDirectory, $TestConfigurationXmlFile) {
 	$SetupTypeXMLs = Get-ChildItem -Path "$WorkingDirectory\XML\VMConfigurations\*.xml"
 	$AllLisaTests = @()
 	$ARMImage = $ARMImageName.Trim().Split(" ")
-	if ( $TestCategory -eq "All") { $TestCategory = "" }
```
Comment (lizzha) [inline Libraries/XmlProcessing.psm1]: These are changed in another refactor PR, please double check where to place these line is better

**Comment by lizzha** [comment]
@ader1990 it's required by Simon to enable running test by test priority. 
Now we have most test cases with priority defined. Chris will add priority for the rest that are imported later.

File: Libraries/XmlProcessing.psm1 (hunk 1 above)
Comment (xian123) [reply Libraries/XmlProcessing.psm1]: Actually, the $TestCategory, $TestArea, etc. are **global** variables and they are not changed the **global** value just by $varName = value format.   That is, their values are "All" not null in CollectTestCases() if they are changed values out of CollectTestCases() via $varName = value.

**Comment by LiliDeng** [comment]
@ader1990 @lizzha Thanks for review, any other comments?

**Comment by juhlee-microsoft** [comment]
@xian123 do you have test result with TestPriority? I am excited to seeing it.

**Pull request merged**